lib/node/init.mjs

```diff
@@ -3,5 +3,8 @@ import Init from "../../dist/node/init.mjs";
 const { main } = Init({ log: "info" });
 
 const root = process.argv[2] || process.cwd();
-const ret = await main(root);
-process.exit(ret ? 0 : 1);
+const p = main(root).then((ret) => {
+  process.exit(ret ? 0 : 1);
+});
+
+Promise.all([p]);
```

Comment:
Hi Alan. The only difference between the two options is that the first option will resolve the module asynchronously while the fix will resolve the module synchronously. In this case nothing gets executed after loading this module so it does not matter.

I would however remove the line with `Promise.all([p])` because it does not have any effect. You can just ignore the promise returned by the `then` method. It is okay to handle the top level promise of an asynchronous computation outside of the promise API and ignore the returned last promise which is meaning-less.

Reply:
Note that you still end up with untracked promises in the end because `Promise.all` also returns a promise. In short without top level awaits, you need to let go the promise at some point and use `Promise.prototype.then` and/or `Promise.prototype.catch` to handle the final outcome outside the Promise API.